engine/services/model_service.cc

```diff
@@ -226,8 +226,11 @@ cpp::result<DownloadTask, std::string> ModelService::HandleDownloadUrlAsync(
     const std::string& url, std::optional<std::string> temp_model_id,
     std::optional<std::string> temp_name) {
   auto url_obj = url_parser::FromUrlString(url);
-  if (url_obj.has_error()) {
-    return cpp::fail("Invalid url: " + url);
+  if (url_obj.has_error() || url_obj->pathParams.size() < 5) {
+    return cpp::fail(
+        "Invalid url: " + url +
+        ", a valid URL example is: "
+        "https://huggingface.co/cortexso/tinyllama/blob/1b/model.gguf");
   }
 
   if (url_obj->host == kHuggingFaceHost) {
```

Comment:
Should we add validation for host url too? I think we only support `huggingface.co`